### DISCUSSION TITLE
Question regarding Time Sync

### DISCUSSION THREAD

**Discussion by rpourzia** [aqualinkd/AqualinkD#235] [Q&A] [answered]
I am curious about the time sync function. How frequently does it sync the controller with the system (ntp) time?  
Also, have you considered adding an api for "on demand" time sync?

**Comment by sfeakes** [comment, marked as answer]
There are two constants in `aqualink.h` that control this.  By default checks time every 3200 seconds (1 hour), and will re-adjust if more than 120 seconds (2 mins) off.   Obviously you can change these are re-compile.
```
#define TIME_CHECK_INTERVAL  3600
#define ACCEPTABLE_TIME_DIFF 120
```
It uses systemtime for the check, so if you want ntp simply set systemtime from ntpd. (or your preferred method)

There is an API (not documented) that will force the time set.   (this will bypasses the check, so time could be out by 10 seconds and it would still set time)
```
http://aqualinkd-ip/api/set_date_time
```

One thing to note, it takes a long time to set the time on the panel, and the panel has no seconds. So it's kind-a difficult to get the time perfectly synced with system time.

**Comment by ballle98** [comment]
@rpourzia I have not looked closely at the RS time sync function but for the PDA version it does not attempt to set the time closer than 1 minute.  It could possible be improved to delay setting the time until the minute transition.  Then the time might be acurate somewhere within a second.  To validate the time below 1 minute accuracy you would watch for the minute transition and compare it against current time since there is no seconds display.

**Reply by rpourzia** [reply]
Thank you @ballle98!  I appreciate the tip.
